cypress/integration/userCanManageParticipants.feature.js

```diff
@@ -0,0 +1,99 @@
+describe("Admin can create and delete participants", () => {
+  beforeEach(() => {
+    cy.intercept("GET", "**/api/fikas**", {
+      fixture: "fikaList.json",
+      statusCode: 200,
+    });
+    cy.intercept("GET", "**/api/departments**", {
+      fixture: "departmentList.json",
+      statusCode: 200,
+    });
+    cy.visit("/");
+    cy.window().its("store").invoke("dispatch", {
+      type: "SET_CURRENT_USER",
+      payload: true,
+    });
+  });
+
+  describe("successfully", () => {
+    beforeEach(() => {
+      cy.intercept("POST", "**/api/participants**", {
+        body: { message: "Participant succesfully added" },
+        statusCode: 201,
+      });
+
+      cy.intercept("GET", "**/api/departments**", {
+        fixture: "updatedDepartmentList.json",
+        statusCode: 200,
+      });
+
+      cy.get("[data-cy=departments-btn]").click();
+      cy.get("[data-cy=department-table]").within(() => {
+        cy.contains("HR").click();
+      });
+    });
+
+    it("is expected to display a manage participants button", () => {
+      cy.get("[data-cy=manage-participants-btn]").should("be.visible");
+    });
+
+    it("is expected to display an input form when clicked", () => {
+      cy.get("[data-cy=manage-participants-btn]").click();
+      cy.get("[data-cy=add-participant-form]").should("be.visible");
+    });
+
+    describe("when a user fills out the form", () => {
+      beforeEach(() => {
+        cy.get("[data-cy=manage-participants-btn]").click();
+        cy.get("[data-cy=name-input]").type("Ben Smith");
+        cy.get("[data-cy=email-input]").type("Ben@email.com");
+        cy.get("[data-cy=start-date-input]").within(() => {
+          cy.get("input:first").clear().type("17/11/2021");
+        });
+        cy.get("[data-cy=management]").within(() => {
+          cy.get("input:first").click();
+        });
+        cy.get("[data-cy=seniority-level]").click();
+        cy.contains("3").click();
+        cy.get("[data-cy=add-btn]").click();
+        cy.wait(500);
+      });
+      it("is expected that the form shows a message on submit", () => {
+        cy.get("[data-cy=submit-response-toast]").should(
+          "contain.text",
+          "Participant succesfully added"
+        );
+      });
+      it("is expected that the added participant is in the table", () => {
+        cy.get("[data-cy=participant-table]").within(() => {
+          cy.contains("Ben Smith").scrollIntoView().should("be.visible");
+        });
```

Comment:
We added a 2nd feature file called `updatedDepartmetList.json`. It's kind of neat how it re-renders and how Ben Smith gets listed: In the `participants` module we call `Department.index()` after the `POST` request, which in turns updates application state. Since React and Redux are quite smart, they know that a component which uses the particular application state must re-render when that, and only that, particular application state changes, Ben gets added to the page after the Modal closes without us having to trigger a re-render with `useState`. Really cool